crypto/ec/ec_check.c

```diff
@@ -9,9 +9,68 @@
 
 #include "ec_lcl.h"
 #include <openssl/err.h>
+#include <openssl/rand_drbg.h>
+#include <openssl/evp.h>
 
+/*
+ * Security strength is a number associated with the amount of work required to
+ * break a cryptographic algorithm or system. For most operations such as key
+ * generation/key agreement the acceptable range is [112, 256]. For legacy
+ * reasons >= 80 is also allowed for decryption and signature verification
+ * operations.
+ */
+#ifdef FIPS_MODE
+# define OSSL_MIN_SECURITY_STRENGTH 112
+# define OSSL_MIN_SECURITY_STRENGTH_LEGACY_USE 80
+#else
+# define OSSL_MIN_SECURITY_STRENGTH 80
+#endif /* FIPS _MODE */
+#define OSSL_MAX_SECURITY_STRENGTH RAND_DRBG_STRENGTH
+
+static int ossl_check_security_strength(int security_bits, int operation)
+{
+#ifdef FIPS_MODE
+    int min_strength = OSSL_MIN_SECURITY_STRENGTH;
+
+    if (operation == EVP_PKEY_OP_UNDEFINED
+            || (operation & (EVP_PKEY_OP_VERIFY | EVP_PKEY_OP_DECRYPT)) != 0)
+        min_strength = OSSL_MIN_SECURITY_STRENGTH_LEGACY_USE;
+    return security_bits >= min_strength && security_bits <= RAND_DRBG_STRENGTH;
+#endif
+    return security_bits <= RAND_DRBG_STRENGTH;
+}
```

Comment:
I'm wondering if we don't need something that can be set by the user. I think it's useful to be able to set a minimum security strength of 112 when not in FIPS mode. Even when in FIPS mode, it might need to be set to a higher value.